editoast/src/views/paced_train.rs

```diff
@@ -202,19 +210,65 @@ struct SimulationBatchForm {
 )]
 async fn simulation_summary(
     State(AppState {
-        db_pool: _db_pool,
-        valkey: _valkey_client,
-        core_client: _core,
+        db_pool,
+        valkey: valkey_client,
+        core_client,
         ..
     }): State<AppState>,
-    Extension(_auth): AuthenticationExt,
+    Extension(auth): AuthenticationExt,
     Json(SimulationBatchForm {
-        infra_id: _infra_id,
-        electrical_profile_set_id: _electrical_profile_set_id,
-        ids: _paced_train_ids,
+        infra_id,
+        electrical_profile_set_id,
+        ids: paced_train_ids,
     }): Json<SimulationBatchForm>,
 ) -> Result<Json<HashMap<i64, SimulationSummaryResult>>> {
-    todo!();
+    let authorized = auth
+        .check_roles([BuiltinRole::OperationalStudies].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Forbidden.into());
+    }
+
+    let conn = &mut db_pool.get().await?;
+
+    let infra = Infra::retrieve_or_fail(conn, infra_id, || PacedTrainError::InfraNotFound {
+        infra_id,
+    })
+    .await?;
+
+    let paced_trains: Vec<PacedTrain> =
+        PacedTrain::retrieve_batch_or_fail(conn, paced_train_ids, |missing| {
+            PacedTrainError::BatchNotFound {
+                count: missing.len(),
+            }
+        })
+        .await?;
+    let paced_trains_to_ts: Vec<TrainSchedule> = paced_trains
+        .clone()
+        .into_iter()
+        .map(|pt| pt.into_first_occurrence())
+        .collect();
+
+    let simulations = train_simulation_batch(
+        conn,
+        valkey_client,
+        core_client,
+        &paced_trains_to_ts,
+        &infra,
+        electrical_profile_set_id,
+    )
+    .await?;
+
+    // Transform simulations to simulation summary
+    let mut simulation_summaries = HashMap::new();
+    for (paced_train, sim) in paced_trains.into_iter().zip(simulations) {
+        let (sim, _) = sim;
+        let simulation_summary_result = simulation_response(sim);
+        simulation_summaries.insert(paced_train.id, simulation_summary_result);
+    }
```

Comment:
nitpick: not mutable, only one destructuration.

```suggestion
    let simulation_summaries = paced_trains
        .into_iter()
        .zip(simulations)
        .map(|(paced_train, (sim, _))| (paced_train.id, simulation_response(sim)))
        .collect();
```